src/main/java/uk/gov/digital/ho/hocs/casework/migration/api/MigrationStageService.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.digital.ho.hocs.casework.migration.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.digital.ho.hocs.casework.api.CaseworkConstants;
+import uk.gov.digital.ho.hocs.casework.api.dto.CreateStageRequest;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.casework.domain.model.BaseStage;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+import uk.gov.digital.ho.hocs.casework.domain.model.Stage;
+import uk.gov.digital.ho.hocs.casework.domain.repository.StageRepository;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.EVENT;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.STAGE_CREATED;
+
+@Slf4j
+@Service
+public class MigrationStageService {
+
+    private final StageRepository stageRepository;
+    private final AuditClient auditClient;
+    private final MigrationCaseDataService migrationCaseDataService;
+
+    public MigrationStageService(StageRepository stageRepository,
+                                 AuditClient auditClient,
+                                 MigrationCaseDataService migrationCaseDataService) {
+        this.stageRepository = stageRepository;
+        this.auditClient = auditClient;
+        this.migrationCaseDataService = migrationCaseDataService;
+    }
+
+    @Transactional
+    public Stage createStageForClosedCase(UUID caseUUID, String stageType) {
+
+        log.debug("Creating Stage: {} for case {}", stageType, caseUUID);
+        CreateStageRequest createStageRequest = new CreateStageRequest(stageType, null, null, "", null, null);
+
+        Set<Stage> allStages = stageRepository.findAllByCaseUUIDAsStage(caseUUID);
+        Set<Stage> activeStages = allStages.stream().filter(stage -> stage.getTeamUUID() != null).collect(Collectors.toSet());
+        if (activeStages.size() > 1) {
+            log.warn("More than 1 active stage for caseUUID: {}, all active stages will be deactivated", caseUUID);
+        }
+
+        Optional<Stage> existingStageOfRequestedType = allStages.stream().filter(stage -> stage.getStageType().equals(createStageRequest.getType())).findFirst();
+
+        final AtomicBoolean isRecreation = new AtomicBoolean(false);
+        final AtomicBoolean isRecreateOfCurrentlyActive = new AtomicBoolean(false);
+        final AtomicReference<Stage> stageToActivate = new AtomicReference<>(null);
+
+        existingStageOfRequestedType.ifPresentOrElse((Stage stage) -> {
```

Comment:
Neither is this required - as you will be creating and not recreating

Reply:
Removed.